packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -349,8 +385,19 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
   }
 
   @override
-  void rejectGesture(int pointer) {
+  void acceptGesture(int pointer) {
+    // Ignore state 'ready' here because that would happen if this recognizer
+    // won by a sweep.
     if (pointer == primaryPointer && state == GestureRecognizerState.possible) {
+      state = GestureRecognizerState.accepted;
+    }
+  }
+
+  @override
+  void rejectGesture(int pointer) {
+    if (pointer == primaryPointer
+        && (state == GestureRecognizerState.possible ||
+            state == GestureRecognizerState.accepted)) {
```

Comment:
i think we should assert that we're not accepted here. it doesn't make sense to reject when you're accepted.

Reply:
It's the existing behavior since there are no knowledge of acceptance in this class yet. We're moving a step closer in this PR but I haven't changed this behavior.